libcontainerd/remote_unix.go

```diff
@@ -414,6 +414,18 @@ func (r *remote) runContainerdDaemon() error {
 	if err := cmd.Start(); err != nil {
 		return err
 	}
+
+	// unless strictly necessary, do not add anything in between here
+	// as the reaper goroutine below needs to kick in as soon as possible
+	// and any "return" from code paths added here will defeat the reaper
+	// process.
+
+	r.daemonWaitCh = make(chan struct{})
+	go func() {
```

Comment:
Yes, all that matters is that `cmd.Wait()` is invoked at one point for this process, no need to have it synced at all. If `dockerd` dies before that happens, `init` will reap it for us.